_source/_posts/2022-09-09-supabase-vue.md

```diff
@@ -0,0 +1,941 @@
+
+```sh
+npm install -D tailwindcss postcss autoprefixer
+npx tailwindcss init -p
+```
+
+Next, go to the file `src/tailwind.config.js` and replace the code there with the following.
+
+```javascript
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ["./index.html", "./src/**/*.{vue,js,ts,jsx,tsx}"],
+  theme: {
+    extend: {},
+  },
+  plugins: [require("daisyui")],
+};
+```
+
+This code will tell Tailwind to look in the file extensions provided to build its final CSS folder. Here you are also telling Tailwind to use `daisyui` as a plugin.
+
+Next, open `src/assets/main.css` and replace the code there with the following.
+
+```CSS
+@import "./base.css";
+
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+```
+
+This will complete the setup of Tailwind CSS.
+
+### Create the Supabase client
+
+To access your Supabase project you will need to call `createClient` from the Supabase client library that you installed earlier. Here you will build a small helper to create the client based on the configuration from your `.env` file.
+
+Create a file in the `src` directory called `supabase.js` and add the following code.
+
+```javascript
+import { createClient } from "@supabase/supabase-js";
+
+const supabaseUrl = "https://tfcejjhqlyrlocdrnfsj.supabase.co";
+const supabaseAnonKey =
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InRmY2VqamhxbHlybG9jZHJuZnNqIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NjE0MzgwMTYsImV4cCI6MTk3NzAxNDAxNn0.X8FVYSOKlZoSVhm4-Q-d7FUEsMvQeHlvf2TzObvf2Ho";
+//import.meta.env.VITE_SUPABASE_URL
+//import.meta.env.VITE_SUPABASE_ANON_KEY
+
+export const supabase = createClient(supabaseUrl, supabaseAnonKey);
+```
+
+This will make it a bit easier to obtain a Supabase client from your views.
+
+### Update boilerplate Vue code
+
+Finally, you need to update the code Vue provided before moving forward.
+
+#### Remove unneeded boilerplate
+
+Delete all the files in the `components` directory. You can also delete the files found in the `views` directory.
+
+#### Update package.json to specify your port
+
+Next, open the `package.json` file and replace the `dev` command with the following code.
+
+```sh
+"dev": "vite --port 3000",
+```
+
+This will force your application to run on port 3000 which is the same port you set up for your Login Callbacks for Okta.
+
+#### Update main.js to set up Okta
+
+You'll need to set up the `okta-vue` client to secure your routes and access auth states. Open `src/main.js` and replace the code there with the following.
+
+```javascript
+import { createApp } from "vue";
+import App from "./App.vue";
+import router from "./router";
+
+import { OktaAuth } from "@okta/okta-auth-js";
+import OktaVue from "@okta/okta-vue";
+
+import "./assets/main.css";
+
+const { VITE_OKTA_DOMAIN, VITE_OKTA_CLIENTID } = import.meta.env;
+
+console.log(VITE_OKTA_DOMAIN);
```

Comment:
We should probably remove this

Reply:
[8e1f4973172a1da929827e1c22a3f5552e866273](https://github.com/nickolasfisher/okta-blog/commit/8e1f4973172a1da929827e1c22a3f5552e866273)